developer_docs/order_in_C.md

```diff
@@ -0,0 +1,259 @@
+    free_array(a);
+    return 0;
+}
+```  
+  
+If the data is composed of at least one variable array, we would use a series of copy operations to our `ndarray`.  
+
+Example:  
+
+```python
+if __name__ == "__main__":
+  import numpy as np
+  a = np.array([1, 2, 3])
+  b = np.array([4, 5, 6])
+  c = np.array([a, [7, 8, 9], b])
+```  
+
+Would translate to this (focus on `c` creation):  
+
+```c
+int main()
+{
+    t_ndarray a = {.shape = NULL};
+    t_ndarray b = {.shape = NULL};
+    t_ndarray c = {.shape = NULL};
+    a = array_create(1, (int64_t[]){3}, nd_int64, false, order_c);
+    int64_t array_dummy[] = {1, 2, 3};
+    memcpy(a.nd_int64 + (a.current_length) , array_dummy, 3 * a.type_size);
+    a.current_length += 3;
+    b = array_create(1, (int64_t[]){3}, nd_int64, false, order_c);
+    int64_t array_dummy_0001[] = {4, 5, 6};
+    memcpy(b.nd_int64 + (b.current_length) , array_dummy_0001, 3 * b.type_size);
+    b.current_length += 3;
+    
+    // 'c' ndarray creation starts here, 'c' is [a, [7, 8, 9], b]
+    
+    c = array_create(2, (int64_t[]){3, 3}, nd_int64, false, order_c); // Allocating 'c' ndarray
+    array_copy_data(&c, a); // Copying the first element of 'c'
+    int64_t array_dummy_0002[] = {7, 8, 9}; // Creating an array_dummy with 'c''s second element's literals ([7, 8, 9])
+    memcpy(c.nd_int64 + (c.current_length) , array_dummy_0002, 3 * c.type_size); // Copying the second element to 'c' ndarray, using 'current_length' as an offset
+    c.current_length += 3; // Incrementing 'current_length' to use as an offset
+    array_copy_data(&c, b); // Copying the third element to 'c' ndarray
+    free_array(a);
+    free_array(b);
+    free_array(c);
+    return 0;
+}
+```  
+
+## `order_f` array creation example
+
+For (`order_f`), the process is similar to (`order_c`), but instead of copying our data straight to the destination `ndarray`, we first create an (`order_c`) `temp_ndarray`, then after finishing the copy operations, we copy a transposed version of `temp_ndarray` to our destination `ndarray`.  
+
+Example:
+
+```python
+if __name__ == "__main__":
+  import numpy as np
+  a = np.array([[1, 2, 3], [4, 5, 6]], order="F")
+  print(a[0][0]) output ==> 1
+```  
+
+Would be translated to this:  
+
+```c
+int main()
+{
+    t_ndarray a = {.shape = NULL};
+    a = array_create(2, (int64_t[]){2, 3}, nd_int64, false, order_f); // Allocation the required ndarray
+    t_ndarray temp_array = {.shape = NULL};
+    temp_array = array_create(2, (int64_t[]){2, 3}, nd_int64, false, order_c); // Allocating an order_c temp_array
+    int64_t array_dummy[] = {1, 2, 3, 4, 5, 6}; // array_dummy with our flattened data
+    memcpy(temp_array.nd_int64 + (temp_array.current_length) , array_dummy, 6 * temp_array.type_size); // Copying our array_dummy to our temp ndarray
+    temp_array.current_length += 6; // Usless in this situation
+    array_copy_data(&a, temp_array); // Copying/Transposing temp_array to the required ndarray
```

Comment:
Is the transpose hidden in this function? I don't see it in the code here

Reply:
Kind of, since `array_copy_data` uses uses `strides` and `shape` to copy from `temp_array` to `a`. so `a` *in memory* is a transposed version of `temp_array`.

Reply:
Oh, but `a` is not actually a transposed version of `temp_array` , since only `strides` are different, shape remains the same...
I should make that clear

Reply:
https://github.com/pyccel/pyccel/pull/1128/commits/b9d21141033ab566337d3c504bbd033d3ca34e29

Reply:
I removed the `transpose` thing from all over the docs.

Reply:
Is this conversation resolved?